fcrepo-serialization/src/main/java/org/fcrepo/camel/serialization/SerializationRouter.java

```diff
@@ -0,0 +1,124 @@
+                .maximumRedeliveries("{{error.maxRedeliveries}}")
+                .log("Index Routing Error: ${routeId}");
+
+        /**
+         * Handle Serialization Events
+         */
+        from("{{input.stream}}")
+            .routeId("FcrepoSerialization")
+            .setHeader(FCREPO_IDENTIFIER).header(IDENTIFIER)
+            .filter(not(or(header(FCREPO_IDENTIFIER).startsWith(simple("{{audit.container}}/")),
+                    header(FCREPO_IDENTIFIER).isEqualTo(simple("{{audit.container}}")))))
+            .choice()
+                .when(header(EVENT_TYPE).isEqualTo(REPOSITORY + "NODE_REMOVED"))
+                    .multicast().to("direct:delete.metadata", "direct:delete.binary").endChoice()
+                .otherwise()
+                    .multicast().to("direct:metadata", "direct:binary");
+
+        from("{{serialization.stream}}")
+            .routeId("FcrepoReSerialization")
+            .filter(not(or(header(FCREPO_IDENTIFIER).startsWith(simple("{{audit.container}}/")),
+                    header(FCREPO_IDENTIFIER).isEqualTo(simple("{{audit.container}}")))))
+            .multicast().to("direct:metadata", "direct:binary");
+
+        from("direct:metadata")
+            .routeId("FcrepoSerializationMetadataUpdater")
+            .to("fcrepo:{{fcrepo.baseUrl}}?accept={{serialization.mimeType}}")
+            .log(INFO, LOGGER,
+                    "Serializing object ${headers[CamelFcrepoIdentifier]}")
+            .setHeader(FILE_NAME)
+                .simple("${headers[CamelFcrepoIdentifier]}.{{serialization.extension}}")
+            .log(DEBUG, LOGGER, "filename is ${headers[CamelFileName]}")
+            .to("{{serialization.descriptions}}");
+
+        from("direct:binary")
+            .routeId("FcrepoSerializationBinaryUpdater")
+            .filter().simple("{{serialization.includeBinaries}} == 'true'")
+            .to("fcrepo:{{fcrepo.baseUrl}}?preferInclude=PreferMinimalContainer")
+            .filter().xpath(isBinaryResourceXPath, ns)
+            .log(INFO, LOGGER,
+                    "Writing binary ${headers[CamelFcrepoIdentifier]}")
+            .to("fcrepo:{{fcrepo.baseUrl}}?metadata=false")
+            .setHeader(FILE_NAME).header(FCREPO_IDENTIFIER)
+            .log(DEBUG, LOGGER, "header filename is: ${headers[CamelFileName]}")
+            .to("{{serialization.binaries}}");
+
+        from("direct:delete.metadata")
+            .routeId("FcrepoSerializationMetadataDeleter")
+            .setHeader(EXEC_COMMAND_ARGS).simple(
+                    "{{serialization.descriptions}}${headers[CamelFcrepoIdentifier]}{{serialization.extension}}")
+            .to("exec:rm")
+            .log(INFO, LOGGER,
+                    "Deleting object from serialized backup ${headers[CamelFcrepoIdentifier]}");
+
+        from("direct:delete.binary")
+            .routeId("FcrepoSerializationBinaryDeleter")
+            .filter().xpath(isBinaryResourceXPath, String.class, ns)
```

Comment:
The message body here will not be XML (there will be no message body in delete events), so you will encounter errors. I would recommend removing the `.filter().xpath(...)` line. In the `exec:rm` line, you will need to handle the case where a file doesn't exist. I would suggest adding a `-f` flag in the `EXEC_COMMAND_ARGS` in which case the route is silent about missing or inaccessible files. Alternately, you can write a filter that first checks for file existence before deleting the file.